### PR TITLE
Callbacks

### PR DESCRIPTION
**Pull request by laurens**
Add support for callbacks using `ActiveModel::Callbacks`.
What do you think?

@baschtl @nicsnet @pixelpogo

### CODE REVIEW

File: dynameister.gemspec [hunk 1]
```diff
@@ -21,6 +21,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.add_dependency "aws-sdk",                               "~> 2.0"
+  spec.add_dependency "activemodel",                           "~> 4.0"
```
Comment (baschtl) [inline dynameister.gemspec]: Can somebody explain why we use `>` for active support and `~>` for active model? Should be unified in my opinion.

File: spec/app/models/pet_food_with_callback.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class PetFoodWithCallback
```
Comment (baschtl) [inline spec/app/models/pet_food_with_callback.rb]: Hmm, pet food. :+1:

File: spec/app/models/pet_food_with_callback.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  include Dynameister::Document
+
+  field :valid_until, :datetime
+  field :created_at,  :time
+  field :deliver_at,  :time
+
+  table
```
Comment (baschtl) [inline spec/app/models/pet_food_with_callback.rb]: Is this really necessary? If yes, it seems to be a weird API case. I thought that at least a hash key will be created by default for the `id` attribute.

**Comment by baschtl** [comment]
@laurens Looks cool!

File: README.md [hunk 4]
```diff
@@ -101,6 +101,26 @@ class CompactDisc
...
+```ruby
+class Person
+  include Dynameister::Document
+
+  field :name
+
+  before_save :ensure_name
+
+  def ensure_name
+    self.name ||= "Heinz"
```
Comment (pixelpogo) [inline README.md]: Heinz? ;-)

File: dynameister.gemspec (hunk 1 above)
Comment (pixelpogo) [inline dynameister.gemspec]: We already use Dynameister in a Rails 5 project thus `~>` did not work, see commit message for https://github.com/babbel/dynameister/commit/2ac3e6d4b029e11e85f96973cba8938db77d44cc.

Therefore we should use `>` for active model as well.

File: spec/app/models/pet_food_with_callback.rb (hunk 3 above)
Comment (pixelpogo) [inline spec/app/models/pet_food_with_callback.rb]: Yes, you are right @baschtl, it is not necessary and should be omitted.

**Comment by pixelpogo** [comment]
@laurens This is great! 
:+1:

File: README.md (hunk 4 above)
Comment (laurens) [inline README.md]: I am willing to change it to Fritz.

> Fritz: Ah, Mrs. Van Hoskins. It's so nice to have you back with us.
> Mrs. Van Hoskins: Thank you, Hans.
> Fritz: Fritz.
> Mrs. Van Hoskins: What happened to Hans?
> Fritz: There is no Hans, Mrs. Van Hoskins, there is only me, Fritz.
> Mrs. Van Hoskins: Oh, what a shame!
> Fritz: [ringing his bell] Boy!
> Mrs. Van Hoskins: Now Franz, I'm going to take this with me. There are some things I need tonight, and tomorrow I want you to put it in the hotel safe for me.

**Comment by nicsnet** [comment]
+B

**Pull request merged**